src/Workspaces/Core/Portable/Shared/Extensions/ICodeDefinitionFactoryExtensions.cs

```diff
@@ -88,7 +88,7 @@ internal static partial class ICodeDefinitionFactoryExtensions
 
             yield return CodeGenerationSymbolFactory.CreateConstructorSymbol(
                 attributes: default(ImmutableArray<AttributeData>),
-                accessibility: Accessibility.Public,
+                accessibility: containingTypeOpt.IsAbstractClass() ? Accessibility.Protected : Accessibility.Public,
```

Comment:
say the user has something like this ```class Base {  public Base(int i) { } }  class Derived : Base { } }```.  Then the user does "new Derived(0)".  We will create a constructor in Derived that takes an int and we will delegate to the base class constructor.